### PR TITLE
Edit GA4 Auto Metric Query & Add Audit Logging to Auto Metrics

### PR DESCRIPTION
**Pull request by mknowlton89**
This PR makes two changes to the auto generate metrics job.

1. It updates the default metric query for GA4 data sources to include an additional where clause using the `_TABLE_SUFFIX` column.

2. It adds audit logging for when an auto metric is created. This looks to be a net-new pattern. Currently, it looks as those most of our audit logging happens directly within controllers. However, we don't create the auto metrics in the controller, but instead kick off an async job. As a result, we can't call `req.audit` in the controller, as we don't have necessary info, like `metric.id` as that has not yet been generated.

As such, we're now having to get the user info and pass that into the job as well. Open to suggestions if there is a way we can abstract the logic so it's less verbose.

### Testing

- [x] Go to a data source that supports auto generated metrics, create 1 or many auto metrics and confirm that there are records for each metric in the `audits` collection in Mongo.
- [x] Go to the metric(s) created and run the query to ensure it works as expected without any errors.

### CODE REVIEW

File: packages/back-end/src/controllers/datasources.ts [hunk 1]
```diff
@@ -437,6 +439,34 @@ export async function putDataSource(
...
+
+  if (!user) {
+    res.status(403).json({
+      status: 403,
+      message: "User not found",
+    });
+    return;
+  }
+
+  const userObj: Omit<
```
Comment (mknowlton89) [inline packages/back-end/src/controllers/datasources.ts:462]: We've discussed removing the ability for a user to add auto metrics from within the `NewDataSourceForm` - once we do that, we can remove this logic.

File: packages/back-end/src/jobs/createAutoGeneratedMetrics.ts [hunk 2]
```diff
@@ -75,6 +84,24 @@ export default function (ag: Agenda) {
         });
 
         await insertMetrics(metrics);
+
+        for (const metric of metrics) {
+          await insertAudit({
```
[the remark this answers is not in the document]
Comment (tzjames) [reply packages/back-end/src/jobs/createAutoGeneratedMetrics.ts:92]: I think it is fine as is for now.  If this is a common need for jobs we could create our own UserJob that extends Job and has an audit function on it.

File: packages/back-end/src/controllers/datasources.ts [hunk 3]
```diff
@@ -437,6 +439,34 @@ export async function putDataSource(
...
+    ExpandedMember,
```
Comment (tzjames) [inline packages/back-end/src/controllers/datasources.ts:463]: Why not make this an `AuditUserLoggedIn`

File: packages/back-end/src/controllers/metrics.ts [hunk 4]
```diff
@@ -342,6 +344,35 @@ export async function postAutoGeneratedMetrics(
...
+
+  if (!user) {
+    res.status(403).json({
+      status: 403,
+      message: "User not found",
+    });
+    return;
+  }
+
+  const userObj: Omit<
```
Comment (tzjames) [inline packages/back-end/src/controllers/metrics.ts:367]: Also here: `AuditUserLoggedIn`

File: packages/back-end/src/jobs/createAutoGeneratedMetrics.ts [hunk 5]
```diff
@@ -23,6 +26,10 @@ type CreateAutoGeneratedMetricsJob = Job<{
     | "dateCreated"
     | "dateUpdated"
   >[];
+  user: Omit<
```
Comment (tzjames) [inline packages/back-end/src/jobs/createAutoGeneratedMetrics.ts:29]: And here:  `AuditUserLoggedIn`

**Review by tzjames — APPROVED** [review]
I had a bunch of minor points, with renaming the event to be `metric.autocreate` being the only substantive one.  I'm approving anyway so I'm not blocking you.

File: packages/back-end/src/integrations/SqlIntegration.ts [hunk 6]
```diff
@@ -1508,7 +1508,7 @@ export default abstract class SqlIntegration
             )}' AND'${formatDate(end, "yyyyMMdd")}'`,
           getAdditionalEvents: () => [],
           getMetricWhereClause: (eventName: string) =>
-            `WHERE event_name = '${eventName}'`,
+            `WHERE _TABLE_SUFFIX BETWEEN '{{date startDateISO "yyyyMMdd"}}' AND '{{date endDateISO "yyyyMMdd"}}' AND event_name = '${eventName}'`,
```
Comment (tzjames) [inline packages/back-end/src/integrations/SqlIntegration.ts:1511]: I have updated this in: https://github.com/growthbook/growthbook/pull/1673 to get the intraday tables as well.  So if I land that first then you will have a merge conflict.  Or you can copy the code from that change.

File: packages/back-end/src/controllers/datasources.ts (hunk 3 above)
Comment (mknowlton89) [reply packages/back-end/src/controllers/datasources.ts:463]: Thank you! I could not find an available type that sufficed!

File: packages/back-end/src/jobs/createAutoGeneratedMetrics.ts [hunk 7]
```diff
@@ -23,6 +27,10 @@ type CreateAutoGeneratedMetricsJob = Job<{
...
```
Comment (tzjames) [inline packages/back-end/src/jobs/createAutoGeneratedMetrics.ts:30]: Looks like you missed this one still.

**Pull request merged**